core/dbt/include/global_project/macros/python_model/python.sql

```diff
@@ -62,11 +62,12 @@ class config:
 
 class this:
     """dbt.this() or dbt.this.identifier"""
-    database = '{{ this.database }}'
-    schema = '{{ this.schema }}'
-    identifier = '{{ this.identifier }}'
+    database = "{{ this.database }}"
+    schema = "{{ this.schema }}"
+    identifier = "{{ this.identifier }}"
+    {% set this_relation_name = this | string | replace('"', '\\"') %}
```

Comment:
One potential option is to make this a macro and have adapters overwrite it

Reply:
This comment is related to the change of respecting quoting schema would break bigquery adapter

Reply:
Just chatted with @colin-rogers-dbt about this. Simplest solution is to turn this processing into a dispatched macro that can be overridden on `dbt-bigquery`.

```
{% macro render_relation_for_python_model(relation) %}
  {{ adapter.dispatch('render_relation_for_python_model', 'dbt')(relation)
{% endmacro %}

{% macro default__render_relation_for_python_model(relation) %}
  {{ return(relation | string | replace('"', \\"') }}
{% endmacro %}

{% macro bigquery__render_relation_for_python_model(relation) %}
  {{ return(relation.quote(database=False, schema=False, identifier=False) | string) }}
{% endmacro %}
```

And use that macro in all three spots that I changed here (`ref`, `source`, `this`)